src/main/java/com/sequenceiq/cloudbreak/service/stack/resource/ResourceBuilder.java

```diff
@@ -2,31 +2,29 @@
 
 import java.util.List;
 
-import com.google.common.base.Optional;
 import com.sequenceiq.cloudbreak.domain.CloudPlatform;
 import com.sequenceiq.cloudbreak.domain.Resource;
 import com.sequenceiq.cloudbreak.domain.ResourceType;
+import com.sequenceiq.cloudbreak.domain.InstanceGroup;
 
-public interface ResourceBuilder<P extends ProvisionContextObject, D extends DeleteContextObject, DCO extends DescribeContextObject,
-        SSCO extends StartStopContextObject> {
+public interface ResourceBuilder<P extends ProvisionContextObject, D extends DeleteContextObject, SSCO extends StartStopContextObject> {
 
-    Boolean create(CreateResourceRequest createResourceRequest) throws Exception;
+    Boolean create(CreateResourceRequest createResourceRequest, InstanceGroup instanceGroup, String region) throws Exception;
 
-    Boolean delete(Resource resource, D deleteContextObject) throws Exception;
+    Boolean delete(Resource resource, D d, String region) throws Exception;
 
-    Boolean rollback(Resource resource, D deleteContextObject) throws Exception;
-
-    Optional<String> describe(Resource resource, DCO describeContextObject) throws Exception;
+    Boolean rollback(Resource resource, D d, String region) throws Exception;
 
     ResourceBuilderType resourceBuilderType();
 
-    Boolean start(SSCO startStopContextObject, Resource resource);
+    Boolean start(SSCO startStopContextObject, Resource resource, String region);
 
-    Boolean stop(SSCO startStopContextObject, Resource resource);
+    Boolean stop(SSCO startStopContextObject, Resource resource, String region);
 
-    List<Resource> buildResources(P provisionContextObject, int index, List<Resource> resources);
+    List<Resource> buildResources(P provisionContextObject, int index, List<Resource> resources, InstanceGroup instanceGroup);
```

Comment:
I think the instanceGroup should be part of the provisionContextObject just like the stack and everything else that's needed to build the request.